go/tools/bazel_testing/bazel_testing.go

```diff
@@ -384,8 +389,9 @@ func setupWorkspace(args Args, files []string) (dir string, cleanup func() error
 
 	// If there's no WORKSPACE file, create one.
 	workspacePath := filepath.Join(mainDir, "WORKSPACE")
-	if _, err := os.Stat(workspacePath); os.IsNotExist(err) {
-		w, err := os.Create(workspacePath)
+	if _, err = os.Stat(workspacePath); os.IsNotExist(err) {
+		var w *os.File
+		w, err = os.Create(workspacePath)
```

Comment:
not quite sure this hunk is needed 🤔

Reply:
are you using some static linter that enforces a certain style?

Reply:
IntelliJ showed me that the `err :=` here shadowed the named return, which breaks the `defer` func below. I fixed this here since I also do this right now in the hunk below, which follows the same pattern.

Reply:
Thanks for the playground link, I didn't know that named returns worked like that.

Reply:
On second thought, I think that this is still problematic: The problem isn't that the `defer` above uses the wrong `err`, it's that the `defer` *below* (line 398-402) sets the wrong `err`, namely the one in the inner scope. Is my thinking correct here?